apps/web/components/dashboard/SortOrderToggle.tsx

```diff
@@ -0,0 +1,64 @@
+import { ButtonWithTooltip } from "@/components/ui/button";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from "@/components/ui/dropdown-menu";
+import { useTranslation } from "@/lib/i18n/client";
+import { api } from "@hoarder/shared-react/trpc";
+import { Check, SortDesc } from "lucide-react";
+import { useRouter, useSearchParams } from "next/navigation";
+
+import type { SortOrder } from "@hoarder/shared/types/bookmarks";
+
+export default function SortOrderToggle() {
+  const { t } = useTranslation();
+  const router = useRouter();
+  const searchParams = useSearchParams();
+  const currentSort = (searchParams.get("sort") as SortOrder) ?? "desc";
+
+  const apiUtils = api.useUtils();
+
+  const updateSort = async (newSort: SortOrder) => {
+    // invalidate the query
+    await apiUtils.bookmarks.getBookmarks.invalidate();
+
+    const params = new URLSearchParams(searchParams.toString());
+    params.set("sort", newSort);
+    router.replace(`?${params.toString()}`);
+
+    // Force a refetch after URL update
+    await apiUtils.bookmarks.getBookmarks.refetch();
+  };
```

Comment:
This and the part that reads the sort order sounds like something that should be extracted into its own shared hook. Specially that as I mentioned later, you'll have to use it for the search endpoint.

Also, the mobile app will eventually have search as well, so I feel like we can make this a zustand store (instead of using search params), and make the hook generic across the mobile and the web. You don't have to do the mobile app part though (I can take care of it).

Reply:
Extracted it into it's own store. Take a look.